File: library/src/main/java/zjm/cst/dhu/library/utils/Router.java
package zjm.cst.dhu.library.utils;

import android.content.Context;
import android.content.Intent;
import android.support.v4.app.Fragment;

/**
 * Created by zjm on 2017/5/11.
 */

public class Router {

    public static Fragment getFragment(String name) {
        Fragment fragment;
        try {
            Class fragmentClass = Class.forName(name);
            fragment = (Fragment) fragmentClass.newInstance();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
        return fragment;
    }

    public static void startActivityForName(Context context, String name) {
        try {
            Class clazz = Class.forName(name);
            Intent intent = new Intent(context, clazz);
            context.startActivity(intent);
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        }
    }
}

File: library/src/main/kotlin/zjm/cst/dhu/library/utils/others/Router.kt
package zjm.cst.dhu.library.utils.others

import android.content.Context
import android.content.Intent
import android.support.v4.app.Fragment

/**
 * Created by zjm on 2017/5/11.
 */

object Router {

    fun getFragment(name: String):
            Fragment? {
        val fragment: Fragment?
        try {
            val fragmentClass = Class.forName(name)
            fragment = fragmentClass.newInstance() as Fragment
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
        return fragment
    }

    fun startActivityForName(context: Context, name: String) {
        try {
            val clazz = Class.forName(name)
            val intent = Intent(context, clazz)
            context.startActivity(intent)
        } catch (e: ClassNotFoundException) {
            throw RuntimeException(e)
        }

    }
}
